_posts/blog/2016-09-06-model-view-viewmodel.md

```diff
@@ -0,0 +1,181 @@
+
+### ViewModel
+
+The ViewModel is a model for the View of the app, an abstraction of the View. The ViewModel retrieves from the DataModel the necessary data, applies the UI logic and then exposes relevant data for the View to consume, together with behaviors for the View. Similar to the DataModel, the ViewModel exposes the data via `Observable`s.
+
+We learned two things about the ViewModel the hard way:
+
+* The ViewModel should expose states for the View, rather than just events. For example, if we need to display the name and the email address of a `User`, rather than creating two streams for this, we create a `DisplayableUser` object that encapsulates the two fields. The stream will emit every time the display name or the email change. Like this, we ensure that our View always displays the current state of the `User`.
+
+* We should make sure that every action of the user goes through the ViewModel and that any possible logic of the View is moved in the ViewModel.
+
+We shared more details about these two topics in a blog post about <a href="https://upday.github.io/mvvm_rx_common_mistakes">common mistakes in MVVM + RxJava</a>.  
+
+### View
+
+The View is the actual user interface in the app. It can be an `Activity`, a `Fragment` or any custom Android `View`. For `Activities` and `Fragment`s, we are binding and unbinding from the event sources on `onResume()` and `onPause()`.
+
+{% highlight java %}
+    private final CompositeSubscription mSubscription = new CompositeSubscription();
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        mSubscription.add(mViewModel.getSomeData()
+                         .observeOn(AndroidSchedulers.mainThread())
+                         .subscribe(this::updateView,
+                                    this::handleError));
+    }
+
+    @Override
+    public void onPause() {
+        mSubscription.clear();
+        super.onPause();
+    }
+{% endhighlight %}
+
+In case the MVVM View is a custom Android `View`, the binding is done in the constructor. To ensure that the subscription is not preserved, leading to possible memory leaks, the unbinding happens in `onDetachedFromWindow`.
+
+{% highlight java %}
+    private final CompositeSubscription mSubscription = new CompositeSubscription();
+
+    public MyView(Context context, MyViewModel viewModel) {
+        ...
+        mSubscription.add(mViewModel.getSomeData()
+                         .observeOn(AndroidSchedulers.mainThread())
+                         .subscribe(this::updateView,
+                                    this::handleError));
+    }
+
+    @Override
+    public void onDetachedFromWindow() {
+        mSubscription.clear();
+        super.onDetachedFromWindow();
+    }
+}
+{% endhighlight %}
+
+## Testability Of The Model-View-ViewModel Classes
+
+One of the main reasons why we love the Model-View-ViewModel pattern is its high degree of testability.
```

Comment:
It depends on the tone you want to use, but "testability" sounds a little stuffy to me if repeated frequently. You could just say:
> it makes <things> so easy to test.

Maybe you don't want to say "easy", but the key point is to not repeat "testability" because it's used in the section title.